src/detect-engine-address.c

```diff
@@ -923,11 +928,7 @@ static int DetectAddressParse2(const DetectEngineCtx *de_ctx,
         } else if (depth == 0 && s[u] == '$') {
             d_set = 1;
         } else if (depth == 0 && u == size - 1) {
-            if (x == sizeof(address)) {
```

Comment:
Why could we have `x == sizeof(address)` and can we no longer have `x == address_length ` ?

Reply:
Good catch.